### ISSUE TITLE
Error on PRIVS: functions

### ISSUE THREAD

**Issue by gabrielpiassetta** [denishpatel/pg-clone-schema#35]
I'm facing this error when trying to clone my schema, probably related with #34 solution

```
ERROR:  Action: PRIVS: Functions  Diagnostics: line=SQL statement "GRANT EXECUTE ON FUNCTION clone_schema_test_b.__crud_attributes(clone_schema_test_b.crud_ops,jsonb,jsonb,jsonb,jsonb) TO "pg_monitor";"
PL/pgSQL function clone_schema(text,text,boolean,boolean) line 729 at EXECUTE. 42883. function clone_schema_test_b.__crud_attributes(clone_schema_test_b.crud_ops, jsonb, jsonb, jsonb, jsonb) does not exist
CONTEXT:  PL/pgSQL function clone_schema(text,text,boolean,boolean) line 773 at RAISE
```

cause the sql that created this function is actually:

```
CREATE OR REPLACE FUNCTION clone_schema_test_b.__crud_attributes(crud_op crud_ops, data jsonb, filtering jsonb DEFAULT NULL::jsonb, sorting jsonb DEFAULT NULL::jsonb, flags jsonb DEFAULT NULL::jsonb)
...
```

to help to debug it, creating the functions using the `src_path_old` as the `search_path` doesn't cause this error.

**Comment by MichaelDBA** [comment]
Yikes.  In reviewing the whole privilege logic, it needs to be totally rewritten using information_schema as the driving force.  Give me a 2-5 days to come up with a new tested solution.

**Comment by MichaelDBA** [comment]
I got this fixed.  Also created another issue#36 to deal with fixing privileges dealing with other objects besides functions.

**Comment by MichaelDBA** [comment]
Got this fixed (Issue#35).  Total rewrite of SQL to get permission stuff 
for functions.  Created anotherissue#36 to do the same kinda rewrites 
for permissions for other objects.  Should have this done as early as 
tomorrow or at least by Tuesday of next week.

Regards,
Michael Vitale


Gabriel Piassetta wrote on 3/4/2021 11:29 AM:
